ql/src/java/org/apache/hadoop/hive/ql/ddl/dataconnector/alter/url/AlterDataConnectorSetUrlOperation.java

```diff
@@ -0,0 +1,65 @@
+/*
```

Comment:
I verified here https://issues.apache.org/jira/browse/HIVE-24396 that
there are no issues with ALTER. In case I missed an already existing
issue, I am really sorry, you can ignore my comment.

I tried the following,

1. Altered a connector to point to the wrong database URL.
2. show tables on an existing database still returns the original tables
3. created a new database with the connector
4. show tables on the new database still returns the original tables.


0: jdbc:hive2://> describe connector extended derby_connector_5;
+--------------------+--------+----------------------+-------------+-------------+----------+----------------------------------------------------+
|        name        |  type  |         url          | owner_name  | owner_type  | comment  |                     parameters                     |
+--------------------+--------+----------------------+-------------+-------------+----------+----------------------------------------------------+
| derby_connector_5  | derby  | jdbc:derby:wrong_db  | narayanan   | USER        |          | {hive.sql.dbcp.password=, hive.sql.dbcp.username=APP} |
+--------------------+--------+----------------------+-------------+-------------+----------+----------------------------------------------------+
1 row selected (0.045 seconds)

0: jdbc:hive2://> describe database db_derby_8;
21/03/26 14:30:38 [47bb12e6-ef2a-4f82-825e-d4d1f1d37148 main]: WARN lazy.LazyStruct: Extra bytes detected at the end of the row! Ignoring similar problems.
+-------------+----------+-----------+------------------+-------------+-------------+--------------------+---------------------+
|   db_name   | comment  | location  | managedlocation  | owner_name  | owner_type  |   connector_name   |    remote_dbname    |
+-------------+----------+-----------+------------------+-------------+-------------+--------------------+---------------------+
| db_derby_8  |          |           |                  | narayanan   | USER        | derby_connector_5  | junit_metastore_db  |
+-------------+----------+-----------+------------------+-------------+-------------+--------------------+---------------------+
1 row selected (0.048 seconds)

0: jdbc:hive2://> use db_derby_8;
No rows affected (0.013 seconds)

0: jdbc:hive2://> show tables;
+-----------+
| tab_name  |
+-----------+
| TBL1      |
| TBL2      |
+-----------+
2 rows selected (0.048 seconds)

0: jdbc:hive2://> CREATE REMOTE DATABASE db_derby_15 USING derby_connector_5 with DBPROPERTIES("connector.remoteDbName"="junit_metastore_db");
21/03/26 15:19:33 [HiveServer2-Background-Pool: Thread-493]: WARN exec.DDLTask: metastore.warehouse.external.dir is not set, falling back to metastore.warehouse.dir. This could cause external tables to use to managed tablespace.
No rows affected (0.018 seconds)

0: jdbc:hive2://> use db_derby_15;
No rows affected (0.015 seconds)

0: jdbc:hive2://> show tables;
+-----------+
| tab_name  |
+-----------+
| TBL1      |
| TBL2      |
+-----------+
2 rows selected (0.056 seconds)

Reply:
Thanks for the verification.